blockchain/api.go

```diff
@@ -1498,6 +1498,27 @@ func (api *BasicEventsAPI) fetchAndProcessLogs(ctx context.Context, filter simpl
 	if err != nil {
 		return err
 	}
+
+	// no logs within the required range,
+	// just push NoEvent data.
+	if len(logs) == 0 {
+		for blockNumber := filter.FromBlock; blockNumber <= filter.ToBlock; blockNumber++ {
+			block, err := api.client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNumber))
+			if err != nil {
+				ctxlog.S(ctx).Warn("failed to get block timestamp", zap.Uint64("block", blockNumber), zap.Error(err))
+				continue
+			}
+
+			receiver <- &Event{
```

Comment:
this could break code which rely on the fact that filter returns event of only one kind. Please check the usage everywhere.

Reply:
Also you can add some parameter to filter to support different types of behaviour